### DISCUSSION TITLE
Custom flux components

### DISCUSSION THREAD

**Discussion by rolfo85** [livewire/flux#931] [Feature Requests]
Is it possible to create custom flux components either from scratch or based on existing components?

the purpose  is to either create brand new components that don't currently exist, or perform modifications to existing components by adding/changing/subtracting core functionalities & behaviours.

**Comment by joshhanley** [comment]
@rolfo85 Flux components are just Blade components, so you could create custom ones by putting them in the Flux folder in `resources/views/flux`, but I wouldn't recommend it. The reason I don't recommend it is because it might cause you issues in future when Flux does create components that might currently be missing. I would recommend just creating your own Blade components (stored in `resources/view/components` and using them instead. Then when Flux comes out with a new component, say like a date picker, you can just do a find and replace of your Blade component to the new Flux component.

But if you still want to do it, you can see how to do it here with custom icons https://fluxui.dev/components/icon#custom-icons

**Reply by rolfo85** [reply]
thanks for the reply. 

I was asking because I would like to use Flux UI to build a design system by wrapping the flux components further.
What is the best approach to wrap the components and add the necessary modifications without altering the original Flux components, as per your recommendation? 

Would the best approach be to copy/paste the code of a Flux component that needs customization, make the required changes, and then wrap it in a Blade component to define the design system?

**Reply by joshhanley** [reply]
@rolfo85 you should just be able to make Blade components and call Flux component inside of them. You should be able to change most of them without needing to publish them, but there might be some that you need to publish. There is a publish console command you can use https://fluxui.dev/docs/customization#publishing-components
